eclair-core/src/main/scala/fr/acinq/eclair/payment/LocalPaymentHandler.scala

```diff
@@ -22,22 +22,23 @@ import fr.acinq.eclair.channel.{CMD_FAIL_HTLC, CMD_FULFILL_HTLC, Channel}
 import fr.acinq.eclair.db.Payment
 import fr.acinq.eclair.payment.PaymentLifecycle.{CheckPayment, ReceivePayment}
 import fr.acinq.eclair.wire._
-
-import scala.concurrent.duration._
 import fr.acinq.eclair.{Globals, NodeParams, randomBytes}
 
 import scala.compat.Platform
 import scala.concurrent.ExecutionContext
-import scala.util.{Failure, Success, Try}
+import scala.concurrent.duration._
+import scala.util.Try
 
 /**
   * Created by PM on 17/06/2016.
   */
-class LocalPaymentHandler(nodeParams: NodeParams)(implicit ec: ExecutionContext = ExecutionContext.Implicits.global) extends Actor with ActorLogging {
+class LocalPaymentHandler(nodeParams: NodeParams) extends Actor with ActorLogging {
+
+  implicit val ec: ExecutionContext = context.system.dispatcher
```

Comment:
You probably noticed that we usually prefer providing an explicit `implicit val ec: ExecutionContext` for future tuning (we may want to execute futures in a separate thread pool from the actor system), but in that case you are right, it is just for the scheduler.